plugins/labhub.py

```diff
@@ -381,3 +382,43 @@ def pr_stats(self, msg, match):
                     state=type(self).community_state(pr_count)
                  )
         yield reply
+
+    @botcmd(split_args_with=None)
+    def ban(self, msg, args):
+        """
+        Ban a user from all Gitter rooms at once.
+        corobo ban <@username/username>
+        """
+        user = msg.frm.nick
+        if not self.TEAMS[self.GH_ORG_NAME + ' maintainers'].is_member(user):
+            yield 'Only maintainers are allowed to use this command.'
+            return
+
+        sinner = args[0]
+        if sinner.startswith('@'):
+            sinner = sinner[1:]
+
+        joined_rooms = self.bot_config.ROOMS_TO_JOIN
+        headers = {'Content-Type': 'application/json',
+                   'Accept': 'application/json',
+                   'Authorization': 'Bearer ' +
+                   self.bot_config.BOT_IDENTITY['token']
+                   }
+        data = json.dumps({"username": sinner})
+
+        r = requests.get('https://api.gitter.im/v1/rooms', headers=headers)
+        room_data = json.loads(r.text)
+        banned_rooms = []
+
+        for room in joined_rooms:
+            for rm in room_data:
+                try:
+                    if rm['uri'] == room:
+                        url = 'https://api.gitter.im/v1/rooms/' + \
+                            rm['id'] + '/bans'
+                        r = requests.post(url, data=data, headers=headers)
+                        banned_rooms.append(room)
+                except KeyError:
+                    pass
```

Comment:
```
for room in filter(lambda x: x['uri'] in joined_rooms, room_data):
   rq = requests.post('https://api.gitter.im/v1/rooms/' + room['id'] + '/bans', data=data, headers=headers)
   banned_users.append(room['uri'])
```
Use something like this, shorter and better ^^ . Not including the exception handling bit, add it ;)

Reply:
sweet